### ISSUE TITLE
Request for training code

### ISSUE THREAD

**Issue by zhicheng2T0** [naver-ai/pit#3]
Hi, would you mind posting the training code for PiT?

**Comment by bhheo** [comment]
Hi

We used [DeiT code](https://github.com/facebookresearch/deit) to train PiT models.
So, you can use it to reproduce PiT results.
Because our training code is exactly the same as [DeiT code](https://github.com/facebookresearch/deit), we will not post it.

**Comment by zhicheng2T0** [comment]
Hi, I am currently trying to recreate PiT-xs under the DeiT training configuration, but the test accuracy increases very slowly during training (at epoch 74, PiT-xs top 1 accuracy is only 25.2%), I am trying to find out if there are any mistakes in my training configuration.

During training, the training hyperparameters I used are: epochs=300, batch size=1024, optimizer=adamp, learning rate=0.001, learning rate decay=cosine, weight decay=0.05, warmup epochs=5, label smoothing epsilon=0.1, no dropout, stochastic depth=0.1, no repeated aug, no gradient clipping, rand augment=9/0.5, mixup prob=0.8, cutmix prob=1.0, erasing prob=0.25. For the model, I directly used the code given in this GitHub repository.

I am wondering if there are any deviations between the hyperparameters I used and the hyperparameters you used when training PiT-xs under DeiT training configurations? Thank you!

**Comment by bhheo** [comment]
Hi

We used the DeiT training code on this commit https://github.com/facebookresearch/deit/tree/ab5715372db8c6cad5740714b2216d55aeae052e

If you use the code in this repo not from the latest timm, it is based on `timm==0.3.4`.
I'm not sure the code in this repo is still compatible with latest timm version.

We put the following arguments to timm code with 4 node on 4 V100 machine.
```
no-repeated-aug:
output_dir: checkpoints
weight-decay: 0.05
lr: 5.00E-04
opt: adamp
batch-size: 256
input-size: 224
```
Some args are overlapped with the default args of DeiT. But, we just put it for clearity.
Please note that timm's `batch-size` means batch-size per GPU.
So, it is 256 batch x 4 GPU = 1024 batch.
Also, deit codebase automatically changes the learning rate. 
https://github.com/facebookresearch/deit/blob/806fb71d37fff07ad6c0fad68d3c8c26683808b6/main.py#L304
So, you don't need to change it for large-batch training.

I don't know what makes the bad performance.
I think the learning rate or the batch-size is a strong suspect for bad performance.

I hope you to success reproducing the performance.

**Comment by guguguzi** [comment]
Hi, I met the same question. Could u please tell me how to train pit by using Deit training code? Thank u!

**Comment by bhheo** [comment]
@guguguzi 
Hi

You can train PiT with DeiT training code https://github.com/facebookresearch/deit/blob/main/main.py
The command is the same as DeiT. https://github.com/facebookresearch/deit#training
Just change the model name to PiT.

For example,
```
python -m torch.distributed.launch --nproc_per_node=4 --use_env main.py --model pit_s_224 --batch-size 256 --data-path /path/to/imagenet
```

If you want to use exactly the same setting then add `--no-repeated-aug --opt adamp`
```
python -m torch.distributed.launch --nproc_per_node=4 --use_env main.py --model pit_s_224 --batch-size 256 --data-path /path/to/imagenet --no-repeated-aug --opt adamp
```

Since `timm` includes PiT and AdamP, you don't need to install it.

**Comment by guguguzi** [comment]
> @guguguzi Hi
> 
> You can train PiT with DeiT training code https://github.com/facebookresearch/deit/blob/main/main.py The command is the same as DeiT. https://github.com/facebookresearch/deit#training Just change the model name to PiT.
> 
> For example,
> 
> ```
> python -m torch.distributed.launch --nproc_per_node=4 --use_env main.py --model pit_s_224 --batch-size 256 --data-path /path/to/imagenet
> ```
> 
> If you want to use exactly the same setting then add `--no-repeated-aug --opt adamp`
> 
> ```
> python -m torch.distributed.launch --nproc_per_node=4 --use_env main.py --model pit_s_224 --batch-size 256 --data-path /path/to/imagenet --no-repeated-aug --opt adamp
> ```
> 
> Since `timm` includes PiT and AdamP, you don't need to install it.

Thanks for your reply! It will help me so much.
